### PR TITLE
[4.0] Replace filter.condition by filter.published in com_content

### PR DESCRIPTION
**Pull request by chmst**
Started here #30459 .

### Summary of Changes
In com_content and related modules, filter.condition is used for filtering publishedm unpublished, archived, trashed data. 
This might cause problems when 3rd party extension set filters for articles. 
This PR sets back the name to filter.published in J3. It does not change the behavour in any way.

### Testing Instructions
Testing is not as easy, as these filters are used in different ways. The test is successful, if nothing in Joomla **frontend** changes with the patch applied. 

You should have some content i.e. testing sample data - categories and articles, some archived, some published, some unpublished and some trashed.
And you should have modules for every kind of articles. 

Now check in the frontend: The modules must show only published articles. Your blogs should show only published articles.

### CODE REVIEW

File: components/com_content/src/Model/CategoriesModel.php [hunk 1]
```diff
@@ -69,7 +69,7 @@ protected function populateState($ordering = null, $direction = null)
 		$params = $app->getParams();
 		$this->setState('params', $params);
 
-		$this->setState('filter.condition',	1);
+		$this->setState('filter.published',	1);
```
Comment (sandewt) [inline components/com_content/src/Model/CategoriesModel.php]: $this->setState('filter.published',	1); should be $this->setState('filter.published', 1);

Comment (chmst) [reply components/com_content/src/Model/CategoriesModel.php]: thanks :)

File: components/com_content/src/Model/CategoriesModel.php [hunk 2]
```diff
@@ -69,8 +69,8 @@ protected function populateState($ordering = null, $direction = null)
 		$params = $app->getParams();
 		$this->setState('params', $params);
 
-		$this->setState('filter.condition',	1);
-		$this->setState('filter.access',	true);
+		$this->setState('filter.published', 1);
```
Comment (Quy) [inline components/com_content/src/Model/CategoriesModel.php:72]: Use `ContentComponent::CONDITION_PUBLISHED` to be consistent elsewhere? If yes, be sure to include `use Joomla\Component\Content\Administrator\Extension\ContentComponent;`.

File: components/com_content/src/Model/CategoryModel.php [hunk 3]
```diff
@@ -169,11 +169,11 @@ protected function populateState($ordering = null, $direction = null)
 		if ((!$user->authorise('core.edit.state', $asset)) &&  (!$user->authorise('core.edit', $asset)))
 		{
 			// Limit to published for people who can't edit or edit.state.
-			$this->setState('filter.condition', 1);
+			$this->setState('filter.published', 1);
```
Comment (Quy) [inline components/com_content/src/Model/CategoryModel.php:172]: Use constant?

File: components/com_content/src/Model/CategoryModel.php [hunk 4]
```diff
@@ -169,11 +169,11 @@ protected function populateState($ordering = null, $direction = null)
...
 		}
 		else
 		{
-			$this->setState('filter.condition', [0, 1]);
+			$this->setState('filter.published', [0, 1]);
```
Comment (Quy) [inline components/com_content/src/Model/CategoryModel.php:176]: Use constants?

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 01c77629595c10bbd3447318ef63a3d7401205ee<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30467">issues.joomla.org/tracker/joomla-cms/30467</a>.</sub>

**Comment by Quy** [comment]
Remove?
https://github.com/joomla/joomla-cms/blob/54d78e9b717b7a8b23a8aeb02817384fdaa6fdd0/modules/mod_articles_latest/src/Helper/ArticlesLatestHelper.php#L69

See:
https://github.com/joomla/joomla-cms/blob/54d78e9b717b7a8b23a8aeb02817384fdaa6fdd0/modules/mod_articles_latest/src/Helper/ArticlesLatestHelper.php#L52

**Comment by karo3** [comment]
I have tested this item :white_check_mark: successfully on 01c77629595c10bbd3447318ef63a3d7401205ee<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30467">issues.joomla.org/tracker/joomla-cms/30467</a>.</sub>

**Comment by richard67** [comment]
@chmst Could you check @Quy 's comment above? Thanks in advance.

**Comment by gostn** [comment]
I have tested this item :white_check_mark: successfully on 01c77629595c10bbd3447318ef63a3d7401205ee<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30467">issues.joomla.org/tracker/joomla-cms/30467</a>.</sub>

**Comment by gostn** [comment]
I have not tested this item.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30467">issues.joomla.org/tracker/joomla-cms/30467</a>.</sub>

**Comment by chmst** [comment]
@Quy could you please remove the "update request" label from this PR, as the requested changes are made in #31155?

**Comment by richard67** [comment]
Previous test results are still valid since last commit was only a clean branch update => RTC.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30467">issues.joomla.org/tracker/joomla-cms/30467</a>.</sub>

**Comment by richard67** [comment]
Good to be merged .. I just gave it a test, too.

**Comment by richard67** [comment]
Thanks!

**Pull request merged**